File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        l = set()
        ans = [0,0]
        for i in range (0,len(grid)):
            for j in range(0,len(grid[0])):
                if grid[i][j] in l:
                    ans[0] = grid[i][j]
                else:
                    l.add(grid[i][j])
        
        n = len(grid)*len(grid[0])
        for i in range(1,n+1):
            if i not in l:
                ans[1]=i

        return ans

File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length*grid[0].length;
        HashSet<Integer> hs = new HashSet<>();
        int[] ans = new int[2];
        for(int i=0;i<grid.length;i++){
            for(int j=0;j<grid[0].length;j++){
                if(hs.contains(grid[i][j])){
                    ans[0] = grid[i][j];
                }else{
                    hs.add(grid[i][j]);
                }
            }
        }
        for(int i=1;i<=n;i++){
            if(!hs.contains(i)){
                ans[1] = i;
            }
        }
        return ans;
    }
}